### PR TITLE
Re-centering for trajectories and Rank-2 fix.

### PR DESCRIPTION
**Pull request by cameronmartino**
#### Bug fixes

* The centering that was added in #27 missed the trajectories. This PR ensures they are centered by using one final check using the means.
* If the rank went below 2 the code meant to add a mock column of zeros for Emperor was not functioning properly. This was fixed.

### CODE REVIEW

File: gemelli/factorization.py [hunk 1]
```diff
@@ -277,6 +277,9 @@ def _fit(self):
             # combine all n_components
             subject_temp_trajectory = np.array(subject_temp_trajectory).T
             feature_temp_trajectory = np.array(feature_temp_trajectory).T
+            # double check check centered
+            subject_temp_trajectory -= subject_temp_trajectory.mean(axis=0)
+            feature_temp_trajectory -= feature_temp_trajectory.mean(axis=0)
```
Comment (gwarmstrong) [inline gemelli/factorization.py]: You should add unit tests to `gemelli.tests.test_factorization` to check if the trajectories are centered.

Comment (cameronmartino) [reply gemelli/factorization.py]: Done -- added to the standalone ctf tests.

**Comment by cameronmartino** [comment]
@gwarmstrong good point on the tests, I added tests for both.

File: gemelli/scripts/tests/test_standalone_ctf.py [hunk 2]
```diff
@@ -56,6 +56,11 @@ def test_standalone_rpca(self):
             index_col=0)
         # Check that the distance matrix matches our expectations
         comp_col = ['PC1', 'PC2', 'PC3']
+        cent_ = samp_res[comp_col].mean().values.max()
+        self.assertLessEqual(cent_, 1e-10)
```
Comment (gwarmstrong) [inline gemelli/scripts/tests/test_standalone_ctf.py]: I would use `assertAlmostEqual(cent_, 0)` for something like this

**Comment by gwarmstrong** [comment]
One more small comment on `assertLessThan(some_value, 1e-7)` vs. `assertAlmostEqual(some_value, 0)`, but it is totally minor and potentially just preference. If you don't want to change this, then I think we're good to go.
https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertAlmostEqual

**Comment by cameronmartino** [comment]
@gwarmstrong good point. done.

**Pull request merged**